mantidproject/mantid
# Add Alias to Generate Python Script as ExportHistory and Add Metadata to Script

**Description of work.**
Added alias to GeneratePythonScript as Export History. Updated the generated script to better format the header and include the Mantid version and the git SHA as metadata. Also created an empty alias for RecordPythonScript as this was causing issues with ExportHistory calling the wrong algorithm due to RecordPythonScript inheriting from GeneratePythonScript. Finally updated GeneratePythonScriptTest to match the new changes.

**To test:**
Run the unit test for **GeneratePythonScript**
Run the below code and check both outputs are the same
```
#create a workspace and run some operations on it
ws = CreateSampleWorkspace()
ws = CropWorkspace(ws, XMin=7828.162291, XMax=11980.906921)
ws = Power(ws, Exponent=1.5)
ws = RenameWorkspace(ws, OutputWorkspace="MyTestWorkspace")

script_text = GeneratePythonScript(ws)
print(script_text.strip())
script_text_2 = ExportHistory(ws)
print(script_text_2.strip())
```

Fixes #27275 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
